Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/controller/AlbumDownloadController.java

```diff
@@ -152,6 +171,43 @@ private void handleDownloadResult(List<ImageSaveTask> tasks) {
         }
     }
 
+    @Override
+    public void onImageProcessed(int downloaded, int failed, int total) {
+        BackgroundUtils.ensureMainThread();
+
+        if (loadingViewController != null) {
+            String message = getAppContext().getString(
+                    R.string.album_download_batch_image_processed_message,
+                    downloaded,
+                    total,
+                    failed
+            );
+
+            loadingViewController.updateWithText(message);
+        }
+    }
+
+    @Override
+    public void onBundleDownloadCompleted() {
+        BackgroundUtils.ensureMainThread();
+
+        if (loadingViewController == null) {
+            throw new IllegalStateException("LoadingViewController is not set!");
+        }
+
+        if (!loadingViewController.shown) {
```

Comment:
Doesn't really matter if the controller is being shown or not because we're getting rid of it anyways. This code can be removed.

Reply:
It's in a case if someone else adds something new that may conflict with the current logic.

Reply:
There's nothing to future-proof here. If you don't toss it out, I will.

Reply:
I still don't get why remove such checks. If you are concerned that it may blow up in release then I can just throw it only in dev builds. These checks are way better than weird bugs with invisible controllers blocking your input so I don't get the point of removing them.

Reply:
The shown variable is basically only internal to the Controller class. It's literally not used anywhere else in the code and it has no effect on whether or not a controller is or is not destroyed. Shown or not, once you call stopPresenting, the shown state is changed to false and the controller is removed. I really should make that variable private.

Reply:
>I really should make that variable private.

Yeah, probably. Well, in this case, yes, there is no point in this check.

Reply:
I guess I misinterpret the whole point of this conversation because I for some reason thought you are talking about all the check (including the null checks) but now I see you were probably only talking about "loadingViewController.shown" check. I have removed it.

Reply:
You've said you've removed it, but the current status of your branch says otherwise. Did you forget to push something?